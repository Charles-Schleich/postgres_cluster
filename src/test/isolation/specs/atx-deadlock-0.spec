setup
{
  create table a();
}

teardown
{
  drop table a;
}

session "s1"
setup         { begin; }
step "s1la"   { lock table a in access exclusive mode; }
step "s1atxb" { begin autonomous; }
step "s1lb"   { lock table a in access exclusive mode; }
step "s1atxc" { commit; }
step "s1c"    { commit; }

permutation "s1la" "s1atxb" "s1lb" "s1atxc" "s1c"
