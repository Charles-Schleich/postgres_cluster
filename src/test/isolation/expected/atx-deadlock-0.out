Parsed test spec with 1 sessions

starting permutation: s1la s1atxb s1lb s1atxc s1c
step s1la: lock table a in access exclusive mode;
step s1atxb: begin autonomous;
step s1lb: lock table a in access exclusive mode;
ERROR:  lockmode 8:AccessExclusiveLock conflicts with itself, acquiring the same lock in that mode again in an autonomous transaction will lead to a sure deadlock
step s1atxc: commit;
step s1c: commit;
